### PR TITLE
Move action buttons to collapsibles with content to review

### PR DESCRIPTION
**Pull request by calvinlu3**
This fixes https://github.com/oncokb/oncokb-pipeline/issues/414

### CODE REVIEW

File: src/main/webapp/app/pages/curation/review/ReviewPage.tsx [hunk 1]
```diff
@@ -40,20 +42,23 @@ const ReviewPage: React.FunctionComponent<IReviewPageProps> = (props: IReviewPag
...
   const [splitView, setSplitView] = useState(false);
 
+  const fetchFirebaseData = () => {
+    // Fetch the data when the user enters review mode. We don't use a listener
+    // because there shouldn't be another user editing the gene when it is being reviewed.
+    get(ref(props.firebaseDb, firebaseGenePath)).then(snapshot => setGeneData(snapshot.val()));
+    get(ref(props.firebaseDb, firebaseMetaReviewPath)).then(snapshot => setMetaReview(snapshot.val()));
+  };
+
   useEffect(() => {
```
Comment (bprize15) [inline src/main/webapp/app/pages/curation/review/ReviewPage.tsx:58]: Probably a slight nit, but you could move the `props.GetDrugs` call to a separate `useEffect`. That way, you could save an API call or two if firebase is not yet ready

File: src/main/webapp/app/pages/curation/review/ReviewPage.tsx [hunk 2]
```diff
@@ -86,28 +90,45 @@ const ReviewPage: React.FunctionComponent<IReviewPageProps> = (props: IReviewPag
...
   };
 
-  const markRootCollapsibleAsPending = (reviewLevelId: string) => {
-    const newReviewChildren = reviewChildren.map(c => {
-      if (c.id === reviewLevelId) {
-        c.hideLevel = true;
-      }
-      return c;
-    });
+  const deleteCollapsible = (reviewLevelId: string, isPending = false) => {
```
Comment (bprize15) [inline src/main/webapp/app/pages/curation/review/ReviewPage.tsx:110]: Maybe either separate into two methods or add a comment so it's clear that this method is called twice

**Review by bprize15 — COMMENTED** [review]
This could be personal preference, but I wonder if we could add the `Created` flag next to the nested cancer type. For me, it seems to improve readability, but I could imagine this not being an issue for curators, so wondering your thoughts?

Screenshot 2024-06-14 at 4 30 27 PM

File: src/main/webapp/app/pages/curation/review/ReviewPage.tsx (hunk 2 above)
Comment (calvinlu3) [reply src/main/webapp/app/pages/curation/review/ReviewPage.tsx]: Separated logic into `deleteCollapsible` and `hideCollapsible`.

**Review by bprize15 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**